libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/EHorizon.kt

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.navigation.base.trip.model
+
+/**
+ * The EHorizon
+ *
+ * @param start
+ */
+data class EHorizon(
+    var start: Edge
+) {
+
+    /**
+     * Get the current edge given the position
+     */
+    fun current(position: EHorizonPosition): Edge {
+        var edge: Edge? = start
+        // TODO Check EHorizonPosition included the Edge and not the edgeId in the original design
+        // while (edge != position.edge) {
+        while (edge?.id != position.edgeId) {
+            edge = edge?.out?.find { e -> e.level == 0 }
+        }
+        return edge
+    }
+
+    /**
+     * Get the current edge given the position
+     */
+    fun trailing(position: EHorizonPosition): List<Edge> {
```

Comment:
Does it collect trailings ONLY along MPP?

Reply:
If the Electronic Horizon module is configured with a `trailingLength` greater than `0` a number of trailing `Edge`s may be collected. This includes the current `Edge` when `position.t` is greater than `0`.

Reply:
Not exposing trailingLength ATM. Might be removed.